client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Internal dependencies
+ */
+import {
+	MEDIA_FILE_UPLOAD_FAILURE,
+	MEDIA_FILE_UPLOAD_SUCCESS,
+	MEDIA_FILE_UPLOADS_ENQUEUE
+} from 'state/action-types';
+import { uploadFile } from 'state/media/actions';
+import { isMediaUploadInProgress, getNextQueuedUpload } from 'state/media/selectors';
+
+export function uploadNext( store ) {
+	const nextQueued = getNextQueuedUpload( store.getState() );
+	if ( nextQueued ) {
+		const { siteId, file } = nextQueued;
+		store.dispatch( uploadFile( siteId, file ) );
+	}
+}
+
+export function maybeUploadFirst( store, action ) {
+	if ( ! isMediaUploadInProgress( store.getState() ) ) {
```

Comment:
Not sure if this is being used elsewhere, but if the value of whether or not we are currently uploading is really contained to this file, does it need to be stored in Redux?

If we kill browser session and reload, do we expect that value to persist and for the app to reinitialize at this point?

If not, I'm open to hiding network-layer-related data here in the network-layer-related files.